frontend/catalyst/compilation_pipelines.py

```diff
@@ -645,11 +648,12 @@ def compute_jvp(self, primals, tangents):
 
         jvps = [jnp.zeros_like(results[res_idx]) for res_idx in range(len(results))]
         for diff_arg_idx, arg_idx in enumerate(argnums):
-            tangent = tangents[arg_idx]
+            tangent = tangents[arg_idx]  # [1]
+            taxis = list(range(tangent.ndim))
             for res_idx in range(len(results)):
                 deriv_idx = diff_arg_idx * len(results) + res_idx
-                num_axes = 0 if tangent.ndim == 0 else 1
-                jvp = jnp.tensordot(jnp.transpose(derivatives[deriv_idx]), tangent, axes=num_axes)
+                deriv = derivatives[deriv_idx]  # [2]
+                jvp = jnp.tensordot(deriv, tangent, axes=(taxis, taxis))
```

Comment:
I'm not sure these formulations are equivalent because we no longer transpose the `[c, d]` dimensions.

I'm not sure which form is "better" (I think our goal here is to produce the same output as JAX), but I think since our Jacobians are transposed compared to those in JAX, at the moment it is necessary to invoke a transpose here. (correct me if I'm wrong)

Reply:
I need to investigate this one. This could be an error we didn't notice before or don't notice now, since `jax.grad` only allows for a scalar results (thus [c,d] becomes []).

Reply:
It's noticeable when you use `jax.jacobian`, I think I'm using it in one test which made me notice it.

Reply:
I have updated the test to operate multi-dimentional outputs and it passed.  So I conclude that the previous version contained error.

Reply:
I have a small flyby question - would it make fixing this bug easier to use `catalyst.jvp` here? 🤔 Or would that be much harder?

I'm just wondering if it makes sense to fix this now, only to replace it again in a few weeks

Reply:
@dime10 what do you think?

Reply:
I mean it's a small fix and the fix is already here, so I don't see why shouldn't merge this is in now.

Reply:
@josh146  adding `catalyst.jvp` could be non-trivial. The calculation is performed on concrete primals (thus no problems with grad) but on _abstract_ tangents. One would need to investigate how to match this API.

Reply:
got it!